src/snapred/backend/recipe/algorithm/PixelGroupingParametersCalculationAlgorithm.py

```diff
@@ -134,21 +137,43 @@ def PyExec(self):
         for groupIndex, groupID in enumerate(groupIDs):
             detIDs = groupingWS.getDetectorIDsOfGroup(int(groupID))
 
+            groupMeanL2 = 0.0
+
             groupMin2Theta = 2.0 * np.pi
             groupMax2Theta = 0.0
-            groupAverage2Theta = 0.0
+            groupMean2Theta = 0.0
+
+            groupMeanPhi = 0.0
+
             pixelCount = 0
             for detID in detIDs:
                 detIndex = detectorInfo.indexOf(int(detID))
                 if detectorInfo.isMonitor(int(detIndex)) or detectorInfo.isMasked(int(detIndex)):
                     continue
+
+                groupMeanL2 += detectorInfo.l2(int(detIndex))
+
+                twoTheta = detectorInfo.twoTheta(int(detIndex))
+                groupMin2Theta = min(groupMin2Theta, twoTheta)
+                groupMax2Theta = max(groupMax2Theta, twoTheta)
+                groupMean2Theta += twoTheta
+
+                try:
+                    groupMeanPhi += detectorInfo.azimuthal(int(detIndex))
```

Comment:
I'm wondering if we shouldn't use
``` python
avgPos = [0,0,0]
for detID in detIDs:
    avgPos += detectorInfo.position()
groupMean2Theta = 2 * asin( avgPos .dotprod. z-axis)
groupMeanPhi = <logic to get phi from the vector>
```

Reply:
I considered something like this.  However, after looking at what `DetectorInfo` was doing to get the angles in the first place, I went with the idea of going with the implementation that modified the existing code as little as possible.
**Eventually, we want to replace `PixelGroupingParametersCalculation` with the equivalent _Mantid_ algorithm anyway.**  (There _is_ something that's called from the GUI that produces almost the same values, but we don't know the specifics yet.  Or, how much difficulty it would be to substitute it in.)